src/SourceBuild/content/build.proj

```diff
@@ -43,6 +53,132 @@
       VmrBuildNumber="$(BUILD_BUILDNUMBER)" />
   </Target>
 
+  <PropertyGroup>
+    <ExtractDotNetSdkArchiveDependsOn>SetupScenarioTestsArtifacts;GetDotNetSdkArtifactPath</ExtractDotNetSdkArchiveDependsOn>
+    <ExtractDotNetSdkArchiveDependsOn Condition="'$(ArchiveExtension)' == '.zip'">$(ExtractDotNetSdkArchiveDependsOn);ExtractDotNetSdkZip</ExtractDotNetSdkArchiveDependsOn>
+    <ExtractDotNetSdkArchiveDependsOn Condition="'$(ArchiveExtension)' == '.tar.gz'">$(ExtractDotNetSdkArchiveDependsOn);ExtractDotNetSdkTarball</ExtractDotNetSdkArchiveDependsOn>
+  </PropertyGroup>
+
+  <Target Name="ExtractDotNetSdkArchive"
+          DependsOnTargets="$(ExtractDotNetSdkArchiveDependsOn)" />
+
+  <PropertyGroup>
+    <_ScenarioTestsNuGetConfig>$(ScenarioTestsArtifactsDir)NuGet.config</_ScenarioTestsNuGetConfig>
+    <_InstallerNuGetConfig>$(RepoRoot)src/installer/NuGet.config</_InstallerNuGetConfig>
+  </PropertyGroup>
+
+  <Target Name="SetupNuGetConfig"
+          Inputs="$(_InstallerNuGetConfig)"
+          Outputs="$(_ScenarioTestsNuGetConfig)">
+
+    <Copy SourceFiles="$(_InstallerNuGetConfig)"
+          DestinationFiles="$(_ScenarioTestsNuGetConfig)" />
+
+    <AddSourceToNuGetConfig NuGetConfigFile="$(_ScenarioTestsNuGetConfig)"
+                            SourceName="local-feed"
+                            SourcePath="$(ScenarioTestsLocalFeedDir)" />
+  </Target>
+
+  <ItemGroup>
+    <_ArtifactsPackage Include="$(ArtifactsShippingPackagesDir)**;
+                                $(ArtifactsNonShippingPackagesDir)**"
+                       Condition="!$([System.String]::Copy('%(Identity)').EndsWith('.symbols.nupkg'))" />
+  </ItemGroup>
+
+  <Target Name="CreateScenarioTestsLocalFeed">
+
+    <RemoveDir Directories="$(ScenarioTestsLocalFeedDir)" />
+    <MakeDir Directories="$(ScenarioTestsLocalFeedDir)" />
+    <!-- Combine all packages into a single directory with symlinks so that we can have a single feed that points to them all. -->
+    <Copy SourceFiles="@(_ArtifactsPackage)"
+          DestinationFolder="$(ScenarioTestsLocalFeedDir)"
+          UseSymbolicLinksIfPossible="true" />
+
+  </Target>
+
+  <Target Name="CreateScenarioTestsDirectoryBuildFiles"
+          Outputs="@(_DirectoryProjFiles)">
+    <!-- Define Directory.Build.* files in the scenario tests artifacts directory to prevent the test run from
+         picking up the configuration from the VMR's Directory.Build.* files. We need an isolated configuration
+         from which to test. -->
```

Comment:
Does MSBuild automatically import `Directory.Build.*` from directories above?

Reply:
MSBuild will scan the directory hierarchy, starting from the project that's being built and moving up through the parents until it finds `Directory.Build.*`. At that point it will stop. We need to prevent it from getting the VMR's definition of this file so this creates stub files which will prevent MSBuild from looking further.